server/src/apub/webfinger.rs

```diff
@@ -0,0 +1,92 @@
+use crate::apub::make_apub_endpoint;
+use crate::db::community::Community;
+use crate::db::establish_connection;
+use crate::db::user::User_;
+use crate::Settings;
+use actix_web::body::Body;
+use actix_web::web::Query;
+use actix_web::HttpResponse;
+use regex::Regex;
+use serde::{Deserialize, Serialize};
+
+#[derive(Deserialize)]
+pub struct WebFingerQuery {
+  resource: String,
+}
+
+#[derive(Default, Serialize, Deserialize)]
+struct WebFingerResponse {
+  subject: String,
+  aliases: Vec<String>,
+  links: Vec<WebFingerLink>,
+}
+
+#[derive(Default, Serialize, Deserialize)]
+struct WebFingerLink {
+  rel: String,
+  r#type: String,
+  href: String,
+}
+
+lazy_static! {
+  static ref WEBFINGER_USER_REGEX: Regex = Regex::new(&format!(
+    "^acct:@([a-zA-Z][0-9a-zA-Z_]*)@{}",
```

Comment:
I dont think this is going to work, the webfinger endpoint on Mastodon doesnt include the initial `@`.

https://radical.town/.well-known/webfinger?resource=acct:felix@radical.town

That means we probably need to use something else instead of `acct:`, which will decrease compatibility with existing implementations. Otherwise we could try to use `acct:` for everything, but that will be problematic if there are a user and a community with the same name. Maybe its still early enough to prevent that in the account/community creation.